### ISSUE TITLE
hlapi cmd with None community causes engine to enter unusable state

### ISSUE THREAD

**Issue by Dylgar** [etingof/pysnmp#191]
When passing None to auth.CommunityData all snmp commands fail with an InconsistentValueError. While I would expect an snmp command to fail with communityIndex of None the error does not make it clear what the issue actually is. This also causes all further requests with the engine produce SmiErrors even to different devices with communityIndexes

The below code is enough to reproduce the error for me running pysnmp v4.4.4. 
```
import pysnmp
import pysnmp.hlapi

target = pysnmp.hlapi.UdpTransportTarget(('localhost', 161), timeout=10)
community_data = pysnmp.hlapi.CommunityData(None)
engine = pysnmp.hlapi.SnmpEngine()
context = pysnmp.hlapi.ContextData()

oi = pysnmp.hlapi.ObjectIdentity('SNMPv2-MIB', ('sysDescr', 0))
ot = pysnmp.hlapi.ObjectType(oi)

results = pysnmp.hlapi.getCmd(engine, community_data, target, context, lookupMib=False, *[ot])

for (errindic, errstat, errindex, binds) in results:
    print (errindic, errstat, errindex, binds)
```

The exact error is:
```
InconsistentValueError: InconsistentValueError({'msg': 'Row consistency check failed for MibScalarInstance((1, 3, 6, 1, 6, 3, 18, 1, 1, 1, 2, 115, 52, 50, 50, 50, 55, 53, 49, 51, 48, 55, 48, 55, 50, 57, 56, 52, 51, 48, 52), <OctetString schema object at 0x7f8f914c49d0 subtypeSpec <ConstraintsIntersection object at 0x7f8f981e1ed0 consts <ValueSizeConstraint object at 0x7f8f981e1e90 consts 0, 65535>> tagSet <TagSet object at 0x7f8f982351d0 tags 0:0:4> encoding iso-8859-1>)'})
```

**Comment by etingof** [comment]
Thank you for raising this issue!

One thing I did not quite understand so far is what are you trying to express by passing `None` as a community name/index. Could you elaborate please?

Assuming you want `None` to act as an empty community string, here's the quick patch that does just that -- #192.

The weird behaviour you've observed is due to lazy computation built inside SNMP engine.

**Comment by Dylgar** [comment]
The devices being queried are from a database and there was some bad data so should have been run against. I would understand receiving an error passing None as the community string. The issue was more around the fact that this caused any further queries run with the snmp engine to fail. Generally with a PyASN1 error.

**Comment by etingof** [comment]
I beg your pardon for taking time on this issue!

Commit ddf30851798e8542bd98bd294785ecb79a5c1ae0 should hopefully fix the root cause of that. The fix is in master (e.g. pysnmp 5) and is going to be released in 4.4.6.

**Comment by etingof** [comment]
jfyi: pysnmp 4.4.6 with this fix has been released
